pywbem_mock/_dmtf_cim_schema.py

```diff
@@ -0,0 +1,426 @@
+        # directory.
+        self._setup_dmtf_schema()
+
+    @property
+    def schema_version(self):
+        """
+        :func:`py:tuple`: of 3 integers with major version, minor version, and
+        update version of the DMTF Schema installed. Ex (2, 49, 0) defines DMTF
+        schema version 2.49.0.
+        """
+        return self._schema_version
+
+    @property
+    def schema_version_str(self):
+        """
+        :term:`string`: with the DMTF CIM schema version in the form
+        <major version>.<minor version>.<revison>.
+        """
+        return '%s.%s.%s' % (self.schema_version[0], self.schema_version[1],
+                             self.schema_version[2])
+
+    @property
+    def schema_root_dir(self):
+        """
+        :term:`string`: the directory in which the DMTF CIM Schema
+        MOF is downloaded and expanded.
+        """
+        return self._schema_root_dir
+
+    @property
+    def schema_mof_dir(self):
+        """
+        :term:`string`: Path name of the directory in which the DMTF CIM Schema
+        MOF files are expanded from the downloaded zip file. This includes the
+        expansion of the individual schema MOF files. This property is useful
+        since it can be used as the MOF compiler search path for compiling
+        classes in the DMTF CIM schema.
+        """
+        return self._schema_mof_dir
+
+    @property
+    def schema_mof_filename(self):
+        """
+        :term:`string`: the path for the DMTF CIM schema MOF top level file
+        which includes the pragmas that define all of the files in the DMTF
+        schema. This is the file that the compiler uses if the complete
+        schema is to be compiled. This filename is of the general form::
+
+            cim_schema<schema_version>.mof
+
+        For example::
+
+            cim_schema_2.49.0.mof
+        """
+        return self._schema_mof_filename
+
+    def __str__(self):
+        """
+        Return a short string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object for human consumption.
+        """
+
+        return '%s(schema_version=%s, schema_root_dir=%s, url=%s)' % \
+               (self.__class__.__name__, self.schema_version_str,
+                self.schema_root_dir, self._mof_zip_url)
+
+    def __repr__(self):
+        """
+        Return a string representation of the
+        :class:`~pywbem_mock.DMTFCIMSchema` object that is suitable for
+        debugging.
+        """
+        return '{0}(schema_version={1}, schema_root_dir={2}, ' \
+               'schema_mof_dir={3}, ' \
+               'schema_mof_filename={4})'.format(self.__class__.__name__,
+                                                 self.schema_version,
+                                                 self.schema_root_dir,
+                                                 self.schema_mof_dir,
+                                                 self.schema_mof_filename)
```

Comment:
I suggest to show all properties in repr, except maybe not the version tuple if we already show the version string.

Reply:
Along with showing the version tuple in str(), showing it here as well is consistent. But there are still other properties missing.